torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2276,50 +2278,70 @@ Tensor eig_backward(const std::vector<torch::autograd::Variable> &grads, const T
   return at::linalg_solve(Uh, at::matmul(U_contrib, Uh) + D_contrib * Uh);
 }
 
-// http://eprints.maths.ox.ac.uk/1079/1/NA-08-01.pdf
-Tensor symeig_backward(const std::vector<torch::autograd::Variable> &grads, const Tensor& self,
-                    bool eigenvectors, bool upper, const Tensor& lambda, const Tensor& v) {
-  // This gradient is symmetric, and not triangular.
-  // symeig operates only on symmetric inputs, which is a subspace of
-  // R^{n x n}, and hence the derivative is not well-defined for off-diagonal
-  // elements. We resolve this by taking the gradient of the functionally independent
-  // elements of the matrix (i.e., the lower triangular portion of the input) and then
-  // reflect it on the upper triangular portion, thereby symmetrizing the gradient of
-  // the symeig operation. The motivation behind this choice is that symmetric gradient
-  // leads to stable gradient updates, and retains symmetry of the updated matrix if it
-  // were updated by a gradient based algorithm.
+// https://people.maths.ox.ac.uk/gilesm/files/NA-08-01.pdf
+Tensor eigh_backward(const std::vector<torch::autograd::Variable> &grads, const Tensor& self,
+                     bool eigenvectors, const Tensor& lambda, const Tensor& v) {
+  // This function is used for both torch.symeig and torch.linalg.eigh.
+  // eigh (and torch.symeig) operates only on symmetric (resp. Hermitian) inputs.
+  // This is a _linear_ subspace of R^{n x n} (resp. C^{n x n}),
+  // since A, B \in Sym(n) => A + B \in Sym(n)
+  // This means that, the tangent space to Sym(n) is Sym(n),
+  // so the gradient is symmetric (resp. Hermitian).
```

Comment:
Is it really true? Note that the matrix `F` below is NOT symmetric, and I am not sure that the numerical argument for the gradient symmetrization holds...